subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/HttpsTaskOutputCachingJavaPerformanceTest.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.regression.buildcache
+
+import org.gradle.performance.fixture.BuildExperimentInvocationInfo
+import org.gradle.performance.fixture.BuildExperimentListener
+import org.gradle.performance.fixture.GradleInvocationSpec
+import org.gradle.performance.fixture.InvocationCustomizer
+import org.gradle.performance.fixture.InvocationSpec
+import org.gradle.test.fixtures.keystore.TestKeyStore
+import spock.lang.Unroll
+
+import static org.gradle.performance.fixture.BuildExperimentRunner.Phase.MEASUREMENT
+import static org.gradle.performance.fixture.BuildExperimentRunner.Phase.WARMUP
+import static org.gradle.performance.regression.buildcache.TaskOutputCachingJavaPerformanceTest.setupTestProject
+import static org.gradle.performance.regression.buildcache.TaskOutputCachingJavaPerformanceTest.touchCacheArtifactsDir
+
+// Gradle profiler doesn't support different parameters per invocation, so we extract the single method here
+@Unroll
+class HttpsTaskOutputCachingJavaPerformanceTest extends AbstractTaskOutputCachingPerformanceTest {
```

Comment:
Let's decide the fate of this test before we merge this PR. I see the following options:

1) delete it: we have coverage for HTTP-only, and we could accept that we don't have coverage for making the HTTPS part of the work somehow slower
2) remove the workaround requiring the custom code, e.g. by creating a trust store that can handle the custom HTTPS server as well as Maven Central

I'd give 2) a go before we fall back to 1).

Reply:
+1.

Let's try 2) first. @blindpirate Can you give that a shot?

Reply:
I fixed it in this PR: https://github.com/gradle/gradle/pull/14383